src/test/java/at/ac/tuwien/kr/alpha/common/fixedinterpretations/FixedInterpretationLiteralsTest.java

```diff
@@ -0,0 +1,199 @@
+	 * case
+	 * where there is output, but no input
+	 */
+	@at.ac.tuwien.kr.alpha.api.externals.Predicate
+	public static Set<List<ConstantTerm<String>>> connection(String dummy) {
+		Set<List<ConstantTerm<String>>> retVal = new HashSet<>();
+		List<ConstantTerm<String>> l1 = new ArrayList<>();
+		List<ConstantTerm<String>> l2 = new ArrayList<>();
+		List<ConstantTerm<String>> l3 = new ArrayList<>();
+		l1.add(ConstantTerm.getInstance("Klagenfurt"));
+		l1.add(ConstantTerm.getInstance("Villach"));
+		l2.add(ConstantTerm.getInstance("Klagenfurt"));
+		l2.add(ConstantTerm.getInstance("Graz"));
+		l3.add(ConstantTerm.getInstance("Villach"));
+		l3.add(ConstantTerm.getInstance("Salzburg"));
+		retVal.add(l1);
+		retVal.add(l2);
+		retVal.add(l3);
+		return retVal;
+	}
+
+	private static final String TEST_PROG = "positive_numeric_comparison :- 2 < 3."
+			+ "negative_numeric_comparison :- not 3 < 2."
+			+ "positive_unary_external :- &isOne[1]."
+			+ "negative_unary_external :- not &isOne[2]."
+			+ "positive_external_with_output :- "
+			+ "&stdlib_datetime_parse[\"01.01.2121 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S), Y = 2121, MO = 1, D = 1, H = 21, MI = 11, S = 12."
+			+ "positive_external_with_output_dontfire :- "
+			+ "&stdlib_datetime_parse[\"01.01.2121 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S), Y = 2345, MO = 1, D = 1, H = 21, MI = 11, S = 12."
+			+ "negative_external_with_output :- Y = 2121, MO = 1, D = 1, H = 21, MI = 11, S = 12, "
+			+ "not &stdlib_datetime_parse[\"22.02.2222 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S)."
+			+ "negative_external_with_output_dontfire :- Y = 2222, MO = 2, D = 22, H = 21, MI = 11, S = 12,"
+			+ "not &stdlib_datetime_parse[\"22.02.2222 21:11:12\", \"dd.MM.yyyy HH:mm:ss\"](Y, MO, D, H, MI, S)."
+			+ "negative_external_multioutput :- START = \"Graz\", END = \"Salzburg\", not &connection[train_network](START, END)."
+			+ "negative_external_multioutput_dontfire :- START = \"Klagenfurt\", END = \"Villach\", not &connection[train_network](START, END)."
+			+ "positive_external_multioutput :- START = \"Klagenfurt\", END = \"Villach\", &connection[train_network](START, END)."
+			+ "positive_external_multioutput_dontfire :- START = \"Graz\", END = \"Salzburg\", &connection[train_network](START, END).";
+
+	private Alpha alpha;
+	private Map<String, PredicateInterpretation> externals;
+
+	public FixedInterpretationLiteralsTest() {
+		this.alpha = new Alpha();
+		this.externals = new HashMap<>();
+		this.externals.putAll(Externals.scan(AspStandardLibrary.class));
+		this.externals.putAll(Externals.scan(FixedInterpretationLiteralsTest.class));
+	}
+
+	@Test
+	public void positiveNumericComparison() {
+		Optional<AnswerSet> answer = this.alpha.solve(this.alpha.readProgramString(TEST_PROG, this.externals)).findFirst();
+		if (!answer.isPresent()) {
+			throw new IllegalStateException("Test setup failed, no answer set!");
+		}
```

Comment:
Just curious: why is this not an `assertTrue(answer.isPresent());" ?

Reply:
Because in the very first version of this class, the whole solving part was in the constructor of the test class (since it's identical for every test case). I only changed that later on when I saw that I couldn't easily step through that code with a debugger when it was part of the constructor. As it stands right now, there's no good reason no to do the assert variant, so I changed that now.